### ISSUE TITLE
GPT-3 model is still being used despite GPT-4 being added in v0.10.0 

### ISSUE THREAD

**Issue by binaryfire** [microsoft/vscode-copilot-release#528]
According to these comments, CoPilot chat should now be powered by GPT-4:
https://github.com/microsoft/vscode-copilot-release/issues/6#issuecomment-1789914587
https://github.com/microsoft/vscode-copilot-release/issues/166#issuecomment-1789923347

However GPT-3 is still being used:
<img width="271" alt="image" src="https://github.com/microsoft/vscode-copilot-release/assets/5361908/1d46882c-19b4-4822-a852-d7f57ebd5e5f">
<img width="451" alt="image" src="https://github.com/microsoft/vscode-copilot-release/assets/5361908/72e8603b-dcba-44bf-9914-f2e4524a3afe">
<img width="393" alt="image" src="https://github.com/microsoft/vscode-copilot-release/assets/5361908/7a92dfdb-698f-4af8-9c94-d6eee8c61182">

Can confirm the answer quality is still GPT-3 level (far lower quality that using GPT-4 via ChatGPT).

**Comment by kylealanhale** [comment]
You can't rely on GPT's self-reporting. This is from the API playground, with no system prompt:

<img width="776" alt="GPT-4 thinks it's GPT-3" src="https://github.com/microsoft/vscode-copilot-release/assets/1770826/b1950225-1fb3-40ca-98bf-4baa9b9a10b2">

**Comment by lramos15** [comment]
Yep I can confirm with 0.10 that GPT-4 is being used via the logs
<img width="913" alt="image" src="https://github.com/microsoft/vscode-copilot-release/assets/4544166/bb83354d-9bf7-48bc-a0ce-0a983490c4b2">

but the LLM doesn't know that 
<img width="391" alt="image" src="https://github.com/microsoft/vscode-copilot-release/assets/4544166/6b540e81-85f4-459b-b40b-1a082f22f7c1">

**Comment by outgaze** [comment]
> Yep I can confirm with 0.10 that GPT-4 is being used via the logs <img alt="image" width="913" src="https://user-images.githubusercontent.com/4544166/280058543-bb83354d-9bf7-48bc-a0ce-0a983490c4b2.png">
> 
> but the LLM doesn't know that <img alt="image" width="391" src="https://user-images.githubusercontent.com/4544166/280058619-6b540e81-85f4-459b-b40b-1a082f22f7c1.png">

Let me rephrase for the user that opened this, yes it does use gpt 4 but it's definitely not using it for the response cause the quality feels as if you are still talking to gpt 3.5 and it's also proven that it still uses that so it's really confusing for the users and feels like there's no difference. 
![gh3](https://github.com/microsoft/vscode-copilot-release/assets/44008063/351bf6b1-8d3f-473b-95d7-f6597a3b295d)

**Comment by binaryfire** [comment]
@outgaze According to this thread, the first response is the GPT-4 powered response to the user prompt. And the second GPT-3.5 response is just the follow up question

https://github.com/microsoft/vscode-copilot-release/issues/6
